apps/console/app/routes/_shell.agent.$agentSlug.branch.$branchSlug.config/form.tsx

```diff
@@ -0,0 +1,270 @@
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@hebo/shared-ui/components/Collapsible";
+import { Input } from "@hebo/shared-ui/components/Input";
+import { Select } from "@hebo/shared-ui/components/Select";
+import { CopyToClipboardButton } from "@hebo/shared-ui/components/code/CopyToClipboardButton";
+import { Split } from "lucide-react";
+
+import { useActionDataErrorToast } from "~console/lib/errors";
+import { FormControl, FormField, FormLabel, FormMessage } from "@hebo/shared-ui/components/Form";
+import { Form } from "react-router";
+import { useState } from "react";
+
+export const ModelConfigSchema = object({
+  models: array(
+    object({
+      alias: pipe(string(), trim(), nonEmpty("Alias is required")),
+      type: pipe(string(), trim(), nonEmpty("Model type is required")),
+    })
+  ),
+});
+
+export type ModelConfigFormValues = InferOutput<typeof ModelConfigSchema>;
+
+const getModelDisplayName = (modelName: string): string => {
+  if (modelName === "custom") return "Custom Model";
+  const model = supportedModels.find((m) => m.name === modelName);
+  return model?.displayName || modelName;
+};
+
+export default function ModelConfigurationForm() {
+  const { agent } = useRouteLoaderData<{
+    agent: { branches: Array<{ models: Array<{ alias: string; type: string }> }> };
+  }>("routes/_shell.agent.$agentSlug")!;
+  
+  const { agentSlug, branchSlug } = useParams<{ agentSlug: string; branchSlug: string }>();
+  const activeBranch = agent.branches[0];
+
+  const actionData = useActionData<any>();
+  useActionDataErrorToast();
+  const navigation = useNavigation();
+  const isSubmitting = navigation.state === "submitting";
+  const currentIntent = String(navigation.formData?.get("intent") || "");
+
+  // Track which items are open/collapsed
+  const [openMap, setOpenMap] = useState<Record<number, boolean>>({});
+
+  // Use Conform as the single source of truth
+  const [form, fields] = useForm<ModelConfigFormValues>({
+    constraint: getValibotConstraint(ModelConfigSchema),
+    defaultValue: { models: activeBranch.models },
+    onValidate({ formData }) {
+      return parseWithValibot(formData, { schema: ModelConfigSchema });
+    },
+    lastResult: actionData,
+    shouldValidate: "onBlur",
+    shouldRevalidate: "onInput",
+  });
+
+  // Get the list of model fields from Conform
+  const modelsList = fields.models.getFieldList();
+
+  const handleAddModel = () => {
+    const currentLength = modelsList.length;
+    form.insert({
+      name: fields.models.name,
+      defaultValue: { alias: "", type: "" },
+    });
+    // Open the newly added item (it will be at the end of the array)
+    setTimeout(() => {
+      setOpenMap((prev) => ({ ...prev, [currentLength]: true }));
+    }, 0);
+  };
+
+  const handleRemoveModel = (index: number) => {
+    form.remove({
+      name: fields.models.name,
+      index,
+    });
+    // Clean up and reindex open state
+    setOpenMap((prev) => {
```

Comment:
This looks very complex, is that really needed / isn't there an easier way to do this?